app/scripts/migrations/045.js

```diff
@@ -0,0 +1,32 @@
+const version = 45
+import { cloneDeep } from 'lodash'
+
+/**
+ * Replace ipfsGateway by 'dweb.link'.
+ */
+export default {
+  version,
+  migrate: async function (originalVersionedData) {
+    const versionedData = cloneDeep(originalVersionedData)
+    versionedData.meta.version = version
+    const state = versionedData.data
+    versionedData.data = transformState(state)
+    return versionedData
+  },
+}
+
+const outdatedGateways = [
+  'ipfs.io',
+  'ipfs.dweb.link',
+]
+
+function transformState (state) {
+  if (!state.PreferencesController) {
+    return state
+  }
+
+  if (!state.PreferencesController.ipfsGateway || outdatedGateways.includes(state.PreferencesController.ipfsGateway)) {
```

Comment:
Can we express this with a single call to `Array#includes`?

```suggestion
  if (outdatedGateways.includes(state?.PreferencesController?.ipfsGateway)) {
```